### DISCUSSION TITLE
API changes / endpoint usage

### DISCUSSION THREAD

**Discussion by Prageeth-Wickramaarachchi** [argoproj/argo-workflows#13687] [Q&A] [answered]
We need to create our own solution using argo workflows as the backend. Doubt I'm having the frequency of the modification of the argo workflow endpoint. Can anyone tell me what should we do?

**Comment by Joibel** [comment]
I'm sorry, but I don't understand your question. Can you explain in more detail what you mean?

**Reply by Prageeth-Wickramaarachchi** [reply]
We are trying to use argo workflows exposed api endpoints. we trying to create a SaaS application where our backend will call to those endpoints to execute workflows. I am asking whether there are continuous changes done on those end points because it might be an issue.

**Reply by Joibel** [reply, marked as answer]
The API endpoints are supposed to be stable, yes. Any breaking changes should be listed in [the upgrading notes](https://argo-workflows.readthedocs.io/en/latest/upgrading/).

The only API change listed since 3.0 is [the change in 3.5](https://argo-workflows.readthedocs.io/en/latest/upgrading/#unified-workflows-list-api-and-ui) to `/api/v1/workflows` where it now also returns archived workflows.

There may be bugs, but we attempt to address them.

**Reply by Prageeth-Wickramaarachchi** [reply]
Can you give me some projects as examples that use argo workflows API endpoints to build their own solutions?

**Reply by MasonM** [reply]
@Prageeth-Wickramaarachchi This is good starting point: https://github.com/search?q=argo+%22%2Fapi%2Fv1%2Fworkflows%22&type=code

**Reply by Prageeth-Wickramaarachchi** [reply]
@MasonM Thanks
